cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -415,11 +416,19 @@ protected Module createModule(CConfiguration cConf, Configuration hConf,
     Module module = new DistributedProgramContainerModule(cConf, hConf, programRunId,
                                                           programOptions.getArguments(), getServiceAnnouncer());
 
-    // In isolated mode, TMS runs in the "launcher" container.
-    // We don't do this in the DistributedProgramContainerModule
-    // because that is used in both launcher and task containers
     if (ProgramRunners.getClusterMode(programOptions) == ClusterMode.ISOLATED) {
-      module = Modules.override(module).with(new MessagingServerRuntimeModule().getStandaloneModules());
+      // In isolated mode, TMS runs in the "launcher" container.
+      // We don't do this in the DistributedProgramContainerModule
+      // because that module is used in both launcher and task containers
+      module = Modules.override(module).with(
+        new MessagingServerRuntimeModule().getStandaloneModules(),
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            // Bind a Cancellable for the RuntimeMonitor to stop this program run.
+            bind(Cancellable.class).toInstance(() -> stop());
```

Comment:
Calling `stop()` in the RuntimeMonitor will stop the program on the remote cluster?

Reply:
it won't. Calling `RuntimeMonitor.kill()` does.